src/wizard/__tests__/wizard.test.tsx

```diff
@@ -515,6 +515,83 @@ describe('Custom actions', () => {
   });
 });
 
+describe('Custom primary actions', () => {
+  test('renders custom primary actions instead of default buttons', () => {
+    const customActions = (
+      <>
+        <Button data-testid="custom-cancel">Custom Cancel</Button>
+        <Button data-testid="custom-next" variant="primary">
+          Custom Next
+        </Button>
+      </>
+    );
+    const [wrapper] = renderDefaultWizard({ customPrimaryActions: customActions });
+
+    expect(wrapper.findPrimaryButton()).toBeNull();
+    expect(wrapper.findCancelButton()).toBeNull();
+    expect(wrapper.findPreviousButton()).toBeNull();
+    expect(wrapper.findActions()!.findButton('[data-testid="custom-cancel"]')).not.toBeNull();
+    expect(wrapper.findActions()!.findButton('[data-testid="custom-next"]')).not.toBeNull();
+  });
+
+  test('custom primary actions work on all steps', () => {
+    const onCustomClick = jest.fn();
+    const customActions = (
+      <Button data-testid="custom-action" onClick={onCustomClick}>
+        Custom Action
+      </Button>
+    );
+
+    const [wrapper, rerender] = renderDefaultWizard({
+      customPrimaryActions: customActions,
+      activeStepIndex: 0,
+    });
+
+    DEFAULT_STEPS.forEach((_, index) => {
+      const customActionButtonWrapper = wrapper.findActions()!.findButton('[data-testid="custom-action"]');
+      expect(customActionButtonWrapper).not.toBeNull();
+      customActionButtonWrapper?.click();
```

Comment:
This should be a strict non-null assertion, `!`. Because otherwise the test silently skips this line